src/napari/_qt/layer_controls/widgets/_vectors/qt_edge_color.py

```diff
@@ -155,21 +156,25 @@ def _on_edge_color_change(self):
                 index = self.color_prop_box.findText(prop, Qt.MatchFixedString)
                 self.color_prop_box.setCurrentIndex(index)
 
-    def _get_property_values(self):
-        """Get the current property values from the Vectors layer
+    def _get_feature_values(self):
+        """Get the current feature values from the Vectors layer
 
         Returns
         -------
-        property_values : np.ndarray
-            array of all of the union of the property names (keys)
-            in Vectors.properties and Vectors.property_choices
+        feature_values : np.ndarray
+            array of all of the union of the property names and choices in
+            Vectors.features (columns in the layer features table)
 
         """
-        property_choices = [*self._layer.property_choices]
-        properties = [*self._layer.properties]
-        property_values = np.union1d(property_choices, properties)
+        feature_choices = [
+            choice
+            for choice, series in self._layer.features.items()
+            if isinstance(series.dtype, pd.CategoricalDtype)
+        ]
```

Comment:
It might be slightly more readable to use [`*self._layer.features.choices`](https://github.com/napari/napari/blob/c751584063b96bba40086e6a0a1252ad7846f7ed/src/napari/layers/utils/layer_utils.py#L816)

But thinking some more, I don't think this method needs to exist anymore. That's because `self._layer.features.columns` will effectively already have this union of feature/property names because it merges the prior behavior of providing `properties` or `property_choices`

Reply:
Thanks for the feedback and useful info! Will give that a check :+1:

Reply:
So gave a check and I think that indeed just using `self._layer.features.columns` should be enough. Also, #7084 has some other worthy changes for the Vectors layer edge control widget related with using/referencing features instead of properties so I basically took the Vectors layer controls related changes from there and added them here im my latest commit ([c56762d](https://github.com/napari/napari/pull/8231/commits/c56762d6e92ae6bd50379d2a702892c26c8d38c1))

Thanks again for pointing out that work!